File: pythondp - banking/service/banking_service.py
from model.account import Account
from factory.account_factory import AccountFactory

class BankingService:

    def __init__(self, repository):
        self.repository = repository

    def create_account(self, user_id, account_type):
        account = Account(user_id, account_type)
        self.repository.save(account)
        return account

    def deposit(self, account_id, amount):
        account = self.repository.find_by_id(account_id)
        account.deposit(amount)

    def withdraw(self, account_id, amount):
        account = self.repository.find_by_id(account_id)
        account.withdraw(amount)

    def create_account(self, user_id, account_type):
        account = AccountFactory.create_account(account_type, user_id)
        self.repository.save(account)
        return account

File: javadp - banking/service/BankingService.java
package service;

import model.Account;
import model.AccountType;
import repository.AccountRepository;
import factory.AccountFactory;

public class BankingService {

    private AccountRepository repository;

    public BankingService(AccountRepository repository) {
        this.repository = repository;
    }

    public Account createAccount(String userId, AccountType type) {
        Account account = new Account(userId, type);
        repository.save(account);
        return account;
    }

    public void deposit(String accountId, double amount) {
        Account account = repository.findById(accountId);
        account.deposit(amount);
    }

    public void withdraw(String accountId, double amount) {
        Account account = repository.findById(accountId);
        account.withdraw(amount);
    }


    public AbstractAccount createAccount(String userId, AccountType type) {
        AbstractAccount account = AccountFactory.createAccount(type, userId);
        repository.save(account);
        return account;
    }
}